### PR TITLE
Markdown: add "open in new window" icon and alt

### LINKED ISSUE #3087: Accessibility: The highlighted links below does not indicate whether they are internal or external links and whether they open in new window or not
image

**Actual**
In LHS screenshot.
- The first link is internal link and opens in a new window. There is no title attribute = “opens in a new window”. Also “opens in a new window” icon is not present.
- The second and third links are external links and opens in a new window. There is no title attribute = “opens in a new window, external”.  
 
**Expected**
Add title attribute as mentioned above and also use icons to represent “external links” and “links opening in new window” 
 
**WCAG Violation**
Predictable
https://www.w3.org/TR/WCAG20-TECHS/G201.html

### PR DESCRIPTION
**Pull request by compulim**
> Fixes #3087.

### Added

-  Resolves [#3087](https://github.com/microsoft/BotFramework-WebChat/issues/3087). External links in Markdown will now appended with an "open in new window" icon and text for accessibility, by [@compulim](https://github.com/compulim) in PR [#3817](https://github.com/microsoft/BotFramework-WebChat/pull/3817)

## Description

For external links in Markdown, adds a `title` attribute to `<a>` and an "open in new window" icon. This is for accessibility.

image

## Design

Updating our Markdown plugins to add `title` attribute and `` for all links.

Instead of inlining the icon in the code, we use a 1 pixel transparent GIF. And use `background-image` in our style set for the "open in new window" icon. This approach allows our end-dev to customize the icon via ideocratic styling.

## Specific Changes

- Updated `renderMarkdown.js` to add `title` attribute and `` for "open in new window" icon
- Added localization strings
- Updated `styleSet/textContent.js` with the "open in new window" icon

-  [x] I have added tests and executed them locally
-  [x] I have updated `CHANGELOG.md`
-  [x] ~I have updated documentation~

## Review Checklist

> This section is for contributors to review your work.

-  [x] Accessibility reviewed (tab order, content readability, alt text, color contrast)
-  [x] Browser and platform compatibilities reviewed
-  [x] CSS styles reviewed (minimal rules, no `z-index`)
-  [x] ~Documents reviewed (docs, samples, live demo)~
-  [x] Internationalization reviewed (strings, unit formatting)
-  [x] ~`package.json` and `package-lock.json` reviewed~
-  [x] ~Security reviewed (no data URIs, check for nonce leak)~
-  [x] Tests reviewed (coverage, legitimacy)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -22,6 +22,10 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 
 ## [Unreleased]
 
+### Added
+
+-  Resolves [#3087](https://github.com/microsoft/BotFramework-WebChat/issues/3087). External links in Markdown will now appended with an "open in new window" icon and text for accessibility, by [@compulim](https://github.com/compulim) in PR [#3817](https://github.com/microsoft/BotFramework-WebChat/pull/3817)
```
Comment (corinagum) [inline CHANGELOG.md:27]:
```suggestion
-  Resolves [#3087](https://github.com/microsoft/BotFramework-WebChat/issues/3087). External links in Markdown will now be appended with an "open in new window" icon and accessibility label, by [@compulim](https://github.com/compulim) in PR [#3817](https://github.com/microsoft/BotFramework-WebChat/pull/3817)
```

File: packages/api/src/localization/en-US.json [hunk 2]
```diff
@@ -65,6 +65,8 @@
   "FILE_CONTENT_DOWNLOADABLE_WITH_SIZE_ALT": "Download file '$1' of size $2",
   "FILE_CONTENT_WITH_SIZE_ALT": "'$1' of size $2",
   "_FILE_CONTENT_ALT.comment": "$1 is the name of the file, e.g. document.docx. $2 is the file size, e.g. 12 MB",
+  "MARKDOWN_EXTERNAL_LINK_ALT": "Opens in a new window, external.",
```
Comment (corinagum) [inline packages/api/src/localization/en-US.json:68]:
```suggestion
  "MARKDOWN_EXTERNAL_LINK_ALT": "Opens in a new window; external.",
```

File: packages/api/src/localization/en-US.json [hunk 3]
```diff
@@ -65,6 +65,8 @@
...
+  "_MARKDOWN_EXTERNAL_LINK_ALT.comment": "This is for screen reader and is the label of the hyperlink, when clicked, will open the hyperlink in a new tab or window.",
```
Comment (corinagum) [inline packages/api/src/localization/en-US.json:69]:
```suggestion
  "_MARKDOWN_EXTERNAL_LINK_ALT.comment": "This is for the screen reader and is added to the label of the hyperlink. When clicked, the hyperlink will open in a new tab or window.",
```

File: packages/component/src/Styles/StyleSet/TextContent.js [hunk 4]
```diff
@@ -16,11 +16,18 @@ export default function createTextContentStyle({ bubbleMaxWidth, bubbleMinHeight
...
...gMjAuNTAwMUw3LjA1NDk1IDIwLjQ5NDlDNS4xNDIzOSAyMC4zOTU0IDMuNjA4OTUgMTguODYyNyAzLjUwODM3IDE2Ljk1MDJMMy41MDMxMiAxNi43NTExTDMuNTAwODkgOC4yNTI3TDMuNTA1MjkgOC4wNTAyQzMuNjA1MzkgNi4xMzc0OSA1LjEzODY3IDQuNjA0NDkgNy4wNTA5NiA0LjUwNTI3TDcuMjUwMSA0LjUwMDE3SDEwLjc0OTVINy4yNTAxWk0xMy43NDgxIDMuMDAxNDZMMjAuMzAxOCAzLjAwMTk3TDIwLjQwMTQgMy4wMTU3NUwyMC41MDIyIDMuMDQzOTNMMjAuNTU5IDMuMDY4MDNDMjAuNjEyMiAzLjA5MTIyIDIwLjY2MzQgMy4xMjE2MyAyMC43MTExIDMuMTU4ODVMMjAuNzgwNCAzLjIyMTU2TDIwLjg2NDEgMy4zMjAxNEwyMC45MTgzIDMuNDEwMjVMMjAuOTU3IDMuNTAwNTdMMjAuOTc2MiAzLjU2NDc2TDIwLjk4OTggMy42Mjg2MkwyMC45OTkyIDMuNzIyODJMMjAuOTk5NyAxMC4yNTU0QzIwLjk5OTcgMTAuNjY5NiAyMC42NjM5IDExLjAwNTQgMjAuMjQ5NyAxMS4wMDU0QzE5Ljg3IDExLjAwNTQgMTkuNTU2MiAxMC43MjMyIDE5LjUwNjUgMTAuMzU3MUwxOS40OTk3IDEwLjI1NTRMMTkuNDk4OSA1LjU2MTQ3TDEyLjI3OTcgMTIuNzg0N0MxMi4wMTM0IDEzLjA1MSAxMS41OTY4IDEzLjA3NTMgMTEuMzAzMSAxMi44NTc1TDExLjIxOSAxMi43ODQ5QzEwLjk1MjcgMTIuNTE4NyAxMC45Mjg0IDEyLjEwMjEgMTEuMTQ2MiAxMS44MDg0TDExLjIxODggMTEuNzI0M0wxOC40MzY5IDQuNTAxNDZIMTMuNzQ4MUMxMy4zNjg0IDQuNTAxNDYgMTMuMDU0NiA0LjIxOTMxIDEzLjAwNSAzLjg1MzI0TDEyLjk5ODEgMy43NTE0NkMxMi45OTgxIDMuMzcxNzcgMTMuMjgwMyAzLjA1Nzk3IDEzLjY0NjQgMy4wMDgzMUwxMy43NDgxIDMuMDAxNDZaIiBmaWxsPSIjMjEyMTIxIiAvPjwvc3ZnPg==)',
```
Comment (corinagum) [inline packages/component/src/Styles/StyleSet/TextContent.js:26]: what about if the user wants to customize this?

Comment (compulim) [reply packages/component/src/Styles/StyleSet/TextContent.js]: Moved to `styleOptions.markdownExternalLinkIconImage`.

**Review by corinagum — APPROVED** [review]

**Pull request merged**